pkg/storage/system_intake_relation.go

```diff
@@ -1,7 +1,71 @@
 package storage
 
 import (
-	_ "embed" // embed is used to embed the SQL file content
+	"context"
+	"errors"
+
+	"github.com/google/uuid"
+	"github.com/jmoiron/sqlx"
+	"go.uber.org/zap"
+
+	"github.com/cmsgov/easi-app/pkg/appcontext"
+	"github.com/cmsgov/easi-app/pkg/models"
 )
 
 // TODO Implement store methods that have to deal with setting System Intake linking/relation data
+func (s *Store) LinkSystemIntakeContractNumbers(ctx context.Context, tx *sqlx.Tx, systemIntakeID uuid.UUID, contractNumbers []string) error {
+	if systemIntakeID == uuid.Nil {
+		return errors.New("unexpected nil system intake ID when linking system intake to contract number")
+	}
+
+	deleteStatement := `
+		DELETE FROM system_intake_contract_numbers
+		WHERE intake_id = $1
+	`
+
+	if _, err := tx.ExecContext(ctx, deleteStatement, systemIntakeID); err != nil {
```

Comment:
I like it for a couple reasons. It's clean (opinion of course), but it also helps in scoping. The `err` only exists in the scope of the `if` block, so it cannot overwrite an outer var (as long as you do `if err :=`, and not `if err =`

Reply:
Whoa. TIL! I didn't realize it changed the scope of `err` -- that's actually pretty huge, because there have been numerous instances where we've come up with really silly error variable names to prevent stomping over each other/reassignment.

Love that!

Reply:
Yeah! Definitely creates safety and avoids issues such as

```
res, err := something()
if err != nil {
        err = doSomethingWhenErrHappens()
        if err != nil {
                log("err during cleanup")
        }
}

return err```

Where the inner func returns `nil` and overwrites the outer err

This is fairly contrived but an actual case that has happened before in our production codebase